sdcm/cluster.py

```diff
@@ -4692,7 +4692,7 @@ def node_setup(self, node: BaseNode, verbose: bool = False, timeout: int = 3600)
                 datacenters = self.datacenter  # pylint: disable=no-member
             SnitchConfig(node=node, datacenters=datacenters).apply()
 
-        if self.params.get('server_encrypt'):
+        if self.params.get('client_encrypt'):
```

Comment:
Change of this condition will break server encryption, as certificates for dbnode-to-dbnode encrypted communication will be created only if 'client_encrypt' is enabled.

Reply:
The problem is indeed in this condition, but on lines:
```
            # Create client facing node certificate, for client-to-node communication
            node.create_node_certificate(
                node.ssl_conf_dir / TLSAssets.DB_CLIENT_FACING_CERT, node.ssl_conf_dir / TLSAssets.DB_CLIENT_FACING_KEY)
```
This is the server certificate, which is facing to client during client-to-dbnode communication.
It should be created in a separate `if self.params.get('client_encrypt'):` condition in this node setup method.

Reply:
Both parts get used in the client encrypt-only setup.
See failed test run here: [scylla-staging/valerii/vp-longevity-aws-custom-d2-workload1-multidc#13](https://argus.scylladb.com/test/2c0a3aad-2dc8-4718-923e-af6271930f70/runs?additionalRuns[]=7ab9b6e8-b782-4786-9c5b-30e831d75440)

Reply:
@dimakr please take over this PR, 

make sure we have a test run to test client / server / client+server

Reply:
Done. The fix turns out to be trivial - as during initializing DB cluster there is a peer check when DB nodes talk to each other for a moment, the server certificates should also be created, not just client-facing ones.